src/zbytes.rs

```diff
@@ -135,8 +125,33 @@ pub unsafe extern "C" fn z_bytes_to_slice(
     dst: &mut MaybeUninit<z_owned_slice_t>,
 ) -> z_result_t {
     let payload = this.as_rust_type_ref();
-    dst.as_rust_type_mut_uninit()
-        .write(payload.to_bytes().into_owned().into());
+    match payload.to_bytes() {
+        std::borrow::Cow::Borrowed(s) => {
+            // force a copy if zbytes is a single shared memory slice
+            #[cfg(all(feature = "unstable", feature = "shared-memory"))]
+            if payload.as_shm().is_some() {
```

Comment:
There is a problem: if compiled as not "unstable" and with "shared-memory" zenoh will still be able to receive SHM buffers, but this check won't be performed. Thus we will have further code possibly executed on SHM buffer.....

Reply:
which is probably fine today, since there is no way to mutate shm buffers without unstable - so getting an increased ref count on shm-slice should not be an issue in this context, right ?